core/trino-spi/src/main/java/io/trino/spi/type/Type.java

```diff
@@ -76,7 +76,7 @@ default TypeOperatorDeclaration getTypeOperatorDeclaration(TypeOperators typeOpe
      * Gets the Java class type used to represent this value on the stack during
      * expression execution.
      * <p>
-     * Currently, this must be boolean, long, double, Slice or Block.
+     * Currently, this can be {@code boolean}, {@code long}, {@code double}, or a non-primitive type.
```

Comment:
Do we accept anything other than `Slice` or `Block`

Reply:
https://github.com/trinodb/trino/blob/03d8e5abc686c5c4a9f96ca14db07e2aed880174/core/trino-spi/src/main/java/io/trino/spi/type/LongTimestampType.java#L49